src/nupic/algorithms/backtracking_tm_cpp.py

```diff
@@ -154,6 +163,44 @@ def __init__(self,
                             )
 
 
+  @classmethod
+  def getSchema(cls):
+    return BacktrackingTMCppProto
+
+
+  def write(self, proto):
+    super(BacktrackingTMCPP, self).write(proto.baseTM)
+    self.cells4.write(proto.cells4)
+    proto.makeCells4Ephemeral = self.makeCells4Ephemeral
+    proto.seed = self.seed
+    proto.checkSynapseConsistency = self.checkSynapseConsistency
+    proto.initArgs = json.dumps(self._initArgsDict)
+
+
+  @classmethod
+  def read(cls, proto):
+    # Use base class to create initial class
+    obj = BacktrackingTM.read(proto.baseTM)
+    obj.__class__ = cls
```

Comment:
Unless you have some concrete argument against it then I think it is the cleanest method. The alternative is to instantiate a new class instance via `object.__new__(cls)` and copy `obj.__dict__` into it. But that results in the full instantiation of an extra temporary class. Or you can create an instance method for reading results into an existing class but that's a very different approach to how the rest of capnp serialization is implemented.